services/org/org.go

```diff
@@ -15,12 +15,19 @@ import (
 	user_model "code.gitea.io/gitea/models/user"
 	"code.gitea.io/gitea/modules/storage"
 	"code.gitea.io/gitea/modules/util"
-	user_service "code.gitea.io/gitea/services/user"
+	repo_service "code.gitea.io/gitea/services/repository"
 )
 
 // DeleteOrganization completely and permanently deletes everything of organization.
-func DeleteOrganization(org *org_model.Organization) error {
-	ctx, commiter, err := db.TxContext(db.DefaultContext)
+func DeleteOrganization(ctx context.Context, org *org_model.Organization, purge bool) error {
+	if purge {
+		err := repo_service.DeleteOwnerRepositoriesDirectly(ctx, org.AsUser())
```

Comment:
```suggestion
		// TODO: Put this into the transaction below, for some reason it currently causes a deadlock (SQLite)
		err := repo_service.DeleteOwnerRepositoriesDirectly(ctx, org.AsUser())
```

Reply:
The deadlock problem was already resolved